lib/broadway_rabbitmq/amqp_client.ex

```diff
@@ -144,6 +144,10 @@ defmodule BroadwayRabbitMQ.AmqpClient do
     |> validate_supported_opts(group, supported)
   end
 
+  defp validate_supported_opts("amqp" <> _ = uri, :connection, _supported_opts) do
+    {:ok, uri}
+  end
```

Comment:
I don't think we should pattern match on the binary, since that might limit what Broadway supports. For example, this AMQP URI is perfectly valid: `"AMQP://127.0.0.1"`.

Since I needed this today I looked into how the validation could work. This is what I came up with:

```elixir
  defp validate_supported_opts(uri, :connection, _) when is_binary(uri) do
    case uri |> to_charlist |> :amqp_uri.parse() do
      {:ok, connection_opts} -> {:ok, connection_opts}
      {:error, reason} -> {:error, "Failed parsing AMQP URI: #{inspect(reason)}"}
    end
  end
```

I considered not doing the validation at all, since `amqp` does it on open anyway. However, I agree that it's leads to a better experience. The error message is nicer and causes the producer to fail early. In contrast, not doing the validation here leads to the producer crashing and attempting to restart multiple times.

Reply:
Yeah over the weekend I wrote a longer validator over here. 

https://github.com/koan-ci/broadway_rabbitmq/pull/1

Would love your input.

Reply:
Also I didn't think about the allcaps version. My main thing was the locally I use amqp due to laziness but in prod I use amqps which has an oddish behaviour which I am attempt to solve in that first draft.

Reply:
@polymetis, I reviewed your approach. I think it's overkill for the purposes of Broadway. It shouldn't be the responsibility of `broadway_rabbitmq` to parse and validate the URI when that is already taken care of by the `amqp` library. For the purposes of Broadway I think a simple error message should suffice stating that the URI couldn't be parsed by `amqp`.

@msaraiva wdyt?

Reply:
A agree. As I stated before, we only try to validate stuff from the underlying client if there's no validation at all and if it's simple enough. We can definitely just say the client can't parse it the URI. As long as we document the different formats (options & URI) it accepts, we are fine.